tests/test_schemes.py

```diff
@@ -2,219 +2,143 @@
 import tempfile
 import os
 
-from wifi import Cell
-from wifi.scheme import extract_schemes, Scheme
 from wifi.exceptions import ConnectionError
-
-
-NETWORK_INTERFACES_FILE = """
-# This file describes the network interfaces available on your system
-# and how to activate them. For more information, see interfaces(5).
-
-# The loopback network interface
-auto lo
-iface lo inet loopback
-
-# The primary network interface
-allow-hotplug eth0
-iface eth0 inet dhcp
-
-iface wlan0-work inet dhcp
-    wpa-ssid workwifi
-    wireless-channel auto
-    wpa-psk 1111111111111111111111111111111111111111111111111111111111111111
-
-iface wlan0-coffee inet dhcp
-    wireless-essid Coffee WiFi
-    wireless-channel auto
-
-iface wlan0-home inet dhcp
-    wpa-ssid homewifi
-    wpa-psk  2222222222222222222222222222222222222222222222222222222222222222
-    wireless-channel auto
-
-iface wlan0-coffee2 inet dhcp
-    wireless-essid Coffee 2
-    wireless-channel auto
-"""
+from dummydata import SchemeTesting as sch
+from wifi.scheme import Scheme
+from wifi import subprocess_compat as subprocess
+from wifi.subprocess_compat import check_output
+from mock import Mock, MagicMock
 
 
 class TestSchemes(TestCase):
     def setUp(self):
         self.tempfile, interfaces = tempfile.mkstemp()
-
         with open(interfaces, 'w') as f:
-            f.write(NETWORK_INTERFACES_FILE)
-
+            f.write(sch.NETWORK_INTERFACES_FILE)
         self.Scheme = Scheme.for_file(interfaces)
 
     def tearDown(self):
         os.remove(self.Scheme.interfaces)
 
     def test_scheme_extraction(self):
-        work, coffee, home, coffee2 = extract_schemes(NETWORK_INTERFACES_FILE)
-
-        assert work.name == 'work'
-        assert work.options['wpa-ssid'] == 'workwifi'
-
-        assert coffee.name == 'coffee'
-        assert coffee.options['wireless-essid'] == 'Coffee WiFi'
+        assert sch.work.name == 'work'
```

Comment:
https://github.com/Erf-/wifi/blob/sudo-args/tests/dummydata.py#L272
I do not think so. I call the extraction function there in the dummydata.